packages/cardano-graphql-services/src/DataProjection/blockHandlers/AssetBlockHandler.ts

```diff
@@ -0,0 +1,124 @@
+      const { assets } = queryResult;
+      const assetIds = assets.map((asset) => asset.assetId);
+      logger.info('About to fetch assets metadata');
+      const metadata = await metadataClient.fetch(assetIds);
+      for (const [index, asset] of assets.entries()) {
+        if (metadata[index]) {
+          asset.tokenMetadata = mapMetadata(metadata[index]);
+        }
+        assetsToInsert.push(asset);
+      }
+    }
+    return { assets: assetsToInsert };
+  },
+  // eslint-disable-next-line sonarjs/cognitive-complexity
+  query: async (ctx: RollForwardContext): Promise<QueryResult> => {
+    const block = getBlockType(ctx.block);
+    let query = '';
+    const assetIdList: string[] = [];
+    if (block?.body !== undefined) {
+      logger.info('About to read block transactions');
+      for (const tx of block.body) {
+        const txBodyMintAssets = tx.body.mint.assets;
+        if (txBodyMintAssets) {
+          logger.info('Assets found');
+          query = `query getAssetsByIds($assetIds: String[]) {
+            queryAsset(filter: { assetId: { in: $assetIds } }) {
+              assetId,
+              assetName,
+              assetNameUTF8,
+              policy,
+              totalQuantity,
+              fingerprint,
+              history,
+              tokenMetadata,
+              nftMetadata
+            }
+          }`;
+          for (const entry of Object.entries(txBodyMintAssets)) {
+            const [policyId, assetName] = entry[0].split('.');
+            assetIdList.push(`${policyId}${assetName !== undefined ? assetName : ''}`);
+          }
+        }
+      }
+    }
+    const variables = assetIdList.length > 0 ? { $assetIds: assetIdList } : {};
+    return { query, variables };
+  },
+  rollBackward: async ({ point }: RollBackwardContext) => {
+    logger.trace('rollBackward', point);
+    const slot = point === 'origin' ? 0 : point.slot;
+    const query = `
+    query {
+      var queryTransaction(func: ge(number, ${slot})) {
+          block{
+	          slot {
+	              number 
+	          }
+          },
+          mint: [
+	          {
+	              Asset AS asset {
+	              assetId,
+	              assetName,
+	              assetNameUTF8,
+	              policy,
+	              totalQuantity,
+	              fingerprint,
+	              history,
+	              tokenMetadata,
+	              nftMetadata
+	              }
+	          }
+          ]
+      }
+    }`;
+    const deleteMutation = {
+      assetId: null,
+      assetName: null,
+      assetNameUTF8: null,
+      fingerprint: null,
+      history: null,
+      nftMetadata: null,
+      policy: null,
+      tokenMetadata: null,
+      totalQuantity: null,
+      uid: 'uid(Asset)'
+    };
+    return {
+      mutations: { ...deleteMutation },
+      variables: { defines: ['Asset'], dql: query }
+    };
+  },
+  rollForward: async (ctx: RollForwardContext, processingResult: CombinedProcessingResult[]) => {
```

Comment:
Shouldn't this receive `CombinedQueryResult`?

Reply:
Isnt the `CombinedProccessingResult` the one that adds the metadata to the asset?

Reply:
Yeah you're right, I thought maybe it could receive both if there is a handler that doesn't have a processing step. Currrent approach is good too: nothing to do.